### DISCUSSION TITLE
run, commit and push status as pre-commit checks?

### DISCUSSION THREAD

**Discussion by indigoviolet** [kevin-hanselman/dud#161] [General]
How does one ensure that the code being checked into git has been executed, committed and pushed with dud?

It would be nice if there were a set of checks that exited with error codes if the "status is dirty" - ie. there are stages that would be run, or not checksummed, or not pushed to the remote

**Comment by kevin-hanselman** [comment]
This is very much an area I want to improve in Dud! I think a `dud status --quiet` flag would be a sensible place to start (which sets an exit code, as you suggest). I'll add an issue for this.
